thor/orbits/iod.py

```diff
@@ -178,8 +184,8 @@ def iod(
     observation_selection_method="combinations",
     iterate=False,
     light_time=True,
-    backend="PYOORB",
-    backend_kwargs={},
+    propagator: Literal["PYOORB"] = "PYOORB",
```

Comment:
Let's think about how to make this open enough that people can install third-party packages and use other propagators.

Reply:
Yes, good idea. In range and transform we pass the actual class, not a string. Maybe we pass the class all the way through to the worker functions and then initialize it in there?

```python
propagator: Type["Propagator"] = PYOORB,
propagator_kwargs: dict = {},
```

Then inside worker functions:
```python
prop = propagator(**propagator_kwargs)
```